connection.go

```diff
@@ -254,68 +234,39 @@ func (co ConnectionOptions) withDefaults() ConnectionOptions {
 	return co
 }
 
-// Creates a new Connection around an outbound connection initiated to a peer
-func (ch *Channel) newOutboundConnection(ctx context.Context, hostPort string, events connectionEvents) (*Connection, error) {
-	timeout := getTimeout(ctx)
-	conn, err := dialContext(ctx, hostPort)
-	if err != nil {
-		if ne, ok := err.(net.Error); ok && ne.Timeout() {
-			ch.log.WithFields(
-				LogField{"remoteHostPort", hostPort},
-				LogField{"timeout", timeout},
-			).Info("Outbound net.Dial timed out.")
-			err = ErrTimeout
-		} else if ctx.Err() == context.Canceled {
-			ch.log.WithFields(
-				LogField{"remoteHostPort", hostPort},
-			).Info("Outbound net.Dial was cancelled.")
-			err = errDialCancelled
-		} else {
-			ch.log.WithFields(
-				ErrField(err),
-				LogField{"remoteHostPort", hostPort},
-			).Info("Outbound net.Dial failed.")
-		}
-		return nil, err
-	}
-
-	return ch.newConnection(conn, hostPort, connectionWaitingToSendInitReq, events), nil
-}
-
-// Creates a new Connection based on an incoming connection from a peer
-func (ch *Channel) newInboundConnection(conn net.Conn, events connectionEvents) (*Connection, error) {
-	return ch.newConnection(conn, "" /* outboundHP */, connectionWaitingToRecvInitReq, events), nil
-}
-
-// Creates a new connection in a given initial state
-func (ch *Channel) newConnection(conn net.Conn, outboundHP string, initialState connectionState, events connectionEvents) *Connection {
+func (ch *Channel) newConnection(conn net.Conn, outboundHP string, remotePeer PeerInfo, remotePeerAddress peerAddressComponents, events connectionEvents) *Connection {
 	opts := ch.connectionOptions.withDefaults()
 
 	connID := _nextConnID.Inc()
 	log := ch.log.WithFields(LogFields{
 		{"connID", connID},
 		{"localAddr", conn.LocalAddr()},
 		{"remoteAddr", conn.RemoteAddr()},
+		{"remoteHostPort", remotePeer.HostPort},
```

Comment:
we could come up with a different name, but this is just a refactoring change, and this was the name of the field previously, so I'd like to avoid any renamings:
https://github.com/uber/tchannel-go/blob/dev/connection.go#L1034